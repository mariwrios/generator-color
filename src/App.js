import React, { useState } from "react"
import { randomColor } from "./helps/randomColor"

import "./index.css"

const App = () => {
  const [cardColor, setCardColor] = useState("")

  const handleColor = () => {
    setCardColor(() => randomColor().length >= 6 && randomColor())
  }
  const color = {
    height: "23vh",
    borderTopRightRadius: 10,
    borderTopLeftRadius: 10,
    width: 250,
    background: `#${cardColor}`
  }
  return (
    <div className="container-card ">
      <div style={color}></div>
      <button onClick={handleColor}>click me</button>
      <p>#{cardColor}</p>
    </div>
  )
}

export default App
