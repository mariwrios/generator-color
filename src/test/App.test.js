import App from "../App"
import React from "react"
import "@testing-library/jest-dom"
import { shallow } from "enzyme"

describe("Probando <App/>", () => {
  const wrapper = shallow(<App />)
  test("should ", () => {
    expect(wrapper).toMatchSnapshot()
  })

  test("Probando que el onClick cambie el estado", () => {
    const p = wrapper.find("p").text()
    wrapper.find("button").simulate("click")
    const p2 = wrapper.find("p").text()
    expect(p).not.toBe(p2)
  })
  test("Probar que llega un valor hexadecimal de 6 y el # digitos al presionar el boton", () => {
    wrapper.find("button").simulate("click")
    const p = wrapper.find("p").text()
    expect(p.length).toBe(7)
    expect(p).toContain("#")
  })
  test("Probando que a la card le llegue el carColor", () => {
    const { style } = wrapper
      .find("div")
      .at(1)
      .props("style")

    expect(style.background.length).toBe(7)
  })
})
