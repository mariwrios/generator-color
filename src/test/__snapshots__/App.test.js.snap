// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` should  1`] = `
<div
  className="container-card "
>
  <div
    style={
      Object {
        "background": "#",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": "23vh",
        "width": 250,
      }
    }
  />
  <button
    onClick={[Function]}
  >
    click me
  </button>
  <p>
    #
  </p>
</div>
`;

exports[`Probando <App/> should  1`] = `
<div
  className="container-card "
>
  <div
    style={
      Object {
        "background": "#",
        "borderTopLeftRadius": 10,
        "borderTopRightRadius": 10,
        "height": "23vh",
        "width": 250,
      }
    }
  />
  <button
    onClick={[Function]}
  >
    click me
  </button>
  <p>
    #
  </p>
</div>
`;
